### ISSUE TITLE
Running tests for stock_landed_costs module raises errors

### ISSUE THREAD

**Issue by wangonya** [odoo/odoo#48205]
### Impacted versions:
- 13

### Steps to reproduce:
- run tests for stock_landed_costs on a new database: 
```
./odoo-bin -i stock_landed_costs --test-enable --test-tags stock_landed_costs -d testdb
```
### Current behavior:
- It seems like the purchase journal is not being found when the tests run.
- Error's raised:
```
INFO testdb odoo.addons.stock_landed_costs.tests.test_stockvaluationlayer: ====================================================================== 
2020-03-23 14:00:07,524 25687 ERROR testdb odoo.addons.stock_landed_costs.tests.test_stockvaluationlayer: ERROR: TestStockValuationLCFIFOVB.test_vendor_bill_flow_anglo_saxon_1
Traceback (most recent call last):
  File "/home/jenga/odoo/addons/stock_landed_costs/tests/test_stockvaluationlayer.py", line 281, in test_vendor_bill_flow_anglo_saxon_1
    vb = Form(self.env['account.move'].with_context(action['context']))
  File "/home/jenga/odoo/odoo/tests/common.py", line 1441, in __init__
    self._init_from_defaults(self._model)
  File "/home/jenga/odoo/odoo/tests/common.py", line 1535, in _init_from_defaults
    for k, v in model.default_get(list(fields)).items()
  File "/home/jenga/odoo/odoo/models.py", line 1205, in default_get
    defaults[name] = field.default(self)
  File "/home/jenga/odoo/addons/account/models/account_move.py", line 77, in _get_default_journal
    raise UserError(error_msg)
odoo.exceptions.UserError: ('Please define an accounting purchase journal in your company', '')
```
```
INFO testdb odoo.addons.stock_landed_costs.tests.test_stockvaluationlayer: ====================================================================== 
2020-03-23 14:00:08,619 25687 ERROR testdb odoo.addons.stock_landed_costs.tests.test_stockvaluationlayer: ERROR: TestStockValuationLCFIFOVB.test_vendor_bill_flow_anglo_saxon_2
Traceback (most recent call last):
  File "/home/jenga/odoo/addons/stock_landed_costs/tests/test_stockvaluationlayer.py", line 369, in test_vendor_bill_flow_anglo_saxon_2
    vb = Form(self.env['account.move'].with_context(action['context']))
  File "/home/jenga/odoo/odoo/tests/common.py", line 1441, in __init__
    self._init_from_defaults(self._model)
  File "/home/jenga/odoo/odoo/tests/common.py", line 1535, in _init_from_defaults
    for k, v in model.default_get(list(fields)).items()
  File "/home/jenga/odoo/odoo/models.py", line 1205, in default_get
    defaults[name] = field.default(self)
  File "/home/jenga/odoo/addons/account/models/account_move.py", line 77, in _get_default_journal
    raise UserError(error_msg)
odoo.exceptions.UserError: ('Please define an accounting purchase journal in your company', '')
```

```
INFO testdb odoo.addons.stock_landed_costs.tests.test_stockvaluationlayer: ====================================================================== 
2020-03-23 14:00:09,685 25687 ERROR testdb odoo.addons.stock_landed_costs.tests.test_stockvaluationlayer: ERROR: TestStockValuationLCFIFOVB.test_vendor_bill_flow_continental_1
Traceback (most recent call last):
  File "/home/jenga/odoo/addons/stock_landed_costs/tests/test_stockvaluationlayer.py", line 421, in test_vendor_bill_flow_continental_1
    vb = Form(self.env['account.move'].with_context(action['context']))
  File "/home/jenga/odoo/odoo/tests/common.py", line 1441, in __init__
    self._init_from_defaults(self._model)
  File "/home/jenga/odoo/odoo/tests/common.py", line 1535, in _init_from_defaults
    for k, v in model.default_get(list(fields)).items()
  File "/home/jenga/odoo/odoo/models.py", line 1205, in default_get
    defaults[name] = field.default(self)
  File "/home/jenga/odoo/addons/account/models/account_move.py", line 77, in _get_default_journal
    raise UserError(error_msg)
odoo.exceptions.UserError: ('Please define an accounting purchase journal in your company', '')
```
- including this module as a dependency on other modules brings up the same issue.

### Expected behavior:
- tests should run with no errors
- the same should happen when the module is included as a dependency

Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
Tests are designed to be execute on installation process, not once you have your database fully populated, and the problem is due to this interaction between 2 modules because of this. This can be obviously improved and there some work in master branch for avoiding some of the technical problems (like DB constraints), but can't be fixed.

Closing as won't fix.

**Comment by wangonya** [comment]
Thanks for your response @pedrobaeza. What do you mean by "not once you have your database fully populated"? I'm running this on a new empty database. The tests fail in the installation process when I include the module as a dependency to another module.

**Comment by pedrobaeza** [comment]
The Odoo test suite is defined to be run all at once and together. There can be some coupling problems (like test failing if there's no such module installed with certain demo data). As said, this is being improved (in v14 there won't be dependency in demo data as well), but for now, this is the best it has. My advice is to run tests only on your modules, not on Odoo core ones.

**Comment by wangonya** [comment]
> My advice is to run tests only on your modules, not on Odoo core ones.

How do I avoid running tests on Odoo core modules when the tests are built into the module? When I inherit the module and include it as a dependency, the tests are run automatically on new builds on Odoo.sh. That's my main issue.

**Comment by pedrobaeza** [comment]
`--test-enable --test-tags <module>`

**Comment by wangonya** [comment]
I thought the code I added when I inherited the module was the one causing the errors so I ran the tests on the module itself just to confirm that everything was working without my customizations. The `--test-enable --test-tags stock_landed_costs` was to prove that.

Here's what is actually happening. 
- I have my module `custom_landed_costs`. 
- Models here inherit from `stock_landed_costs` so in my dependencies, I have `stock_landed_costs` listed. 
- I finish my customizations and push. 
- A new build is triggered on Odoo.sh. 
- Build fails because of said tests from `stock_landed_costs`

So my question is, since these tests are built into the core module, how do I stop them from running on the installation process of the new build on Odoo.sh?
